pkg/datapath/maps/map.go

```diff
@@ -45,84 +42,57 @@ var (
 // This interface is provided to abstract epmanager/filesystem access for unit
 // testing.
 type endpointManager interface {
```

Comment:
It seems reasonable to add `EndpointExists` and `HasGlobalCT` to endpoint manager directly, but then use this interface with those 2 functions to avoid the direct import.
`RemoveDatapathMapping` and `RemoveMapPath` seem very restricted to this package, and could just be utility functions inside it? That would also make it more clear what the thing implementing the package-local `endpointManager ` is. I might be missing something but loading the interface up seems like it only causes more boilerplate. Perhaps we have a plan to move those last two functions into endpoint manager at some point?

Reply:
Perhaps I can split up things to be two interfaces instead of just one to avoid having to wrap endpointmanager and so that I can pass in the EndpointManager directly.

Reply:
Had to look at the code again, but the reason that `RemoveDatapathMapping` are `RemoveMapPath` are part of the interface is that the unit tests have their own implementation of these which does not interact with the filesystem. So, storing them as utility functions would break the testing there (and would require having BPF imports and filesystem interactions).

Reply:
ah, I see... yeah... bummer but 🚢 . It wasn't a big deal in any case.